nemo/core/classes/modelPT.py

```diff
@@ -80,32 +80,23 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None):
         self._trainer = trainer
 
```

Comment:
Also a more fundamental question, why does restore_from() attempt to restore dataloaders?

Reply:
Restore from calls model constrictor, which in turn loads the new or old cfg file, which in turn has the current or old path to the data directory, which in turn crashes resume if run on a new machine. This is a safety net against that. Seems @nithinraok has already faced issues cause of this behaviour.

Reply:
So why do we do this?
Why not just remove all dataloaders from config when restoring?
Why not have a restore_weights_only function?
If the design has flaws, we can change it, instead of hacking around it.

Reply:
yes, my suggestion is to have a separate section in cfg to mention data_related stuff and have a setup_data() method which calls  set_up_{test, train, valid}_data() methods to load dataloaders. 
This way when we save or restore model only save model related stuff and not interfere with data.

Reply:
@okuchaiev Please add points here. 
From my view, suggestion to remove dataloader from serialization means the user has to write their own config after installing model from a .Nemo file cthis defeats the purpose of .Nemo. Instead, the proposed workflow which @okuchaiev is working on is -
1) Load the checkpoint, bypass the incorrect config and show it to the user in the log. 
2) User overwrites the critical path directly in Python (OmegaConf)
3) them call the appropriate setup function which this data loader. We overwrite the self._cfg.*_ds definition and use the override just as always. 

This allows the user 
1) to have a predefined config with almost every value already set
2) to have something that can be overriden in Python rather than mess around with more yaml
3) to safely support packaging all the remaining items - tokemixers, and data files from NLP also fall under this. Not all domains follow the manifest approach, but this allos override by user at any level since they have access to the full cfg after instantiation (well everything other than the trainer config)

Reply:
@nithinraok As we discussed in the call, thus is not a useful solution. NLP, CV don't follow the same rules and structure that ASR, does not does TTS. You are thinking only from the view of ASR. The data loader config would again have to be defined by the user for your solution, entirely defeating the purpose of the packaged .Nemo file

Reply:
As to having a restore_weigits_only, again, Core cannot make assumptions about which parts of the manifest needs paths. Look at NLP manifests, their structure is entirely different than ASR. Look at CV which doesn't even have paths. Why should core delete configs which declare splits of train eval test for a dataset and force the user to redefine it in Yaml or python ?

Reply:
> Load the checkpoint, bypass the incorrect config and show it to the user in the log.

So it seems to me, the answer to should restore_from() attempt to restore dataloaders is: No, restore_from() should not attempt to restore dataloaders.
Isn't that what you are doing? Remove the offending dataloader parts from the config, instantiate the model, and then instantiate the datasets.

Reply:
Yes, in a sense. What I try is case 1) original model owner is training so path is correct (if not, I will print the warning with full stack trace under debug level). They then export .Nemo. 

Case 2) new user on another machine uses .Nemo file to restore in say a Jupyter notebook or script. This raises the warning (but user likely will not use debug level logging so no stack trace shown). 

They get an example of the entire validation of test config that was attempted to load. They can then deepcopy the mldel._cfg and then update model._cfg.*_ds.* to override whatever offending path (or data or tokenizer or split etc). Finally they follow the instructions in the warning and call the appropriate setup method with this updated config, thereby successfully loading their own data + whatever other config changes to the dataset they want.

This way, yes, we safely notify the user that the manifest was incorrect, without making assumptions as to how the manifest is built - this allows cross domain flexibility. 

We show the user the config, so they can edit as they please. This is lost if we destroy the config themselves during save. Restore from also can't make that assumptions as to how data loading will occur across domains. So this simply sidesteps the issue across domains.

Reply:
@blisc  The promise behind the model is that it contains all the necessary details for training/validation/testing.
So "Why not just remove all dataloaders from config when restoring"  - because there are important information in the dataloader config. For example, for ASR models - sample_rate, batch_size, normalize transcripts or not, for BERT whether max token was 128 or 512, etc.

If would be nice if we could keep those details and drop paths to local files/folders. But, unfortunately, those are handled differently from model to model. Hence, this patch.